openverse_catalog/dags/templates/create_provider_ingester.py

```diff
@@ -0,0 +1,152 @@
+
+
+def _render_file(
+    target: Path,
+    template_path: Path,
+    provider: str,
+    endpoint: str,
+    media_types: list[str],
+    name: str,
+):
+    with target.open("w", encoding="utf-8") as target_file:
+        filled_template = _get_filled_template(
+            template_path, provider, endpoint, media_types
+        )
+        target_file.write(filled_template)
+        print(f"{name + ':':<18} {target.relative_to(PROJECT_PATH)}")
+
+
+def fill_template(provider, endpoint, media_types):
+    print(f"Creating files in {REPO_PATH}")
+    provider = provider.replace(" ", "_")
+
+    dags_path = TEMPLATES_PATH.parent / "providers"
+    api_path = dags_path / "provider_api_scripts"
+    filename = inflection.underscore(provider)
+
+    # Render the API file itself
+    script_template_path = TEMPLATES_PATH / "template_provider.py_template"
+    api_script_path = api_path / f"{filename}.py"
+    _render_file(
+        api_script_path,
+        script_template_path,
+        provider,
+        endpoint,
+        media_types,
+        "API script",
+    )
+
+    # Render the tests
+    script_template_path = TEMPLATES_PATH / "template_test.py_template"
+    tests_path = REPO_PATH / "tests"
+    # Mirror the directory structure, but under the "tests" top level directory
+    test_script_path = tests_path.joinpath(*api_path.parts[-3:]) / f"test_{filename}.py"
+    _render_file(
+        test_script_path,
+        script_template_path,
+        provider,
+        endpoint,
+        media_types,
+        "API script test",
+    )
+
+    print(
+        """
+NOTE: You will also need to add a new ProviderWorkflow dataclass configuration to the \
+PROVIDER_WORKFLOWS list in `openverse-catalog/dags/providers/provider_workflows.py`.
+"""
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Create a new provider API ProviderDataIngester",
+        add_help=True,
+    )
+    parser.add_argument(
+        "provider", help='Create the ingester for this provider (eg. "Wikimedia").'
+    )
+    parser.add_argument(
+        "endpoint",
+        help="API endpoint to fetch data from"
+        ' (eg. "https://commons.wikimedia.org/w/api.php").',
+    )
+    parser.add_argument(
+        "-m",
+        "--media",
+        type=str,
+        nargs="*",
+        help="Ingester will collect media of these types"
+        " ('audio'/'image'). Default value is ['image',]",
+    )
+    args = parser.parse_args()
+    provider = args.provider
+    endpoint = args.endpoint
+
+    # Get valid media types
+    media_types = []
+    for media_type in args.media:
+        if media_type in MEDIA_TYPES:
+            media_types.append(media_type)
+        else:
+            print(f"Ignoring invalid type {media_type}")
+
+    # Default to image if no valid media types given
+    if not media_types:
+        print('No media type given, defaulting to ["image",]')
+        media_types = [
+            "image",
+        ]
```

Comment:
When I tried the script without specifying the media type I got an error. This way should be fixed to `image` as default:

```suggestion
    media_types = []

    # Default to image if no valid media types given
    if not args.media:
        print('No media type given, defaulting to ["image",]')
        media_types = [
            "image",
        ]
    else:
        # Get valid media types
        for media_type in args.media:
            if media_type in MEDIA_TYPES:
                media_types.append(media_type)
            else:
                print(f"Ignoring invalid type {media_type}")
```

Reply:
Thank you for catching this! I've pulled this out into a separate method and unit tested it as well :)